### PR TITLE
Added Yoneda, Coyoneda, Cayley, Composition, Sieve, Cosieve

### PR DESCRIPTION
**Pull request by xgrommx**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@
 /node_modules/
 /output/
 package-lock.json
+.DS_Store
```
Comment (hdgarrood) [inline .gitignore:8]: I think it’s better in general to add entries like this to your global gitignore file, rather in each individual project’s gitignore

**Review by hdgarrood — COMMENTED** [review]
I’m hesitant about additions like this, because a) it’s not entirely clear to me when these things would be useful, and b) if they ever need to change in a breaking way, then that will have downstream effects for quite a lot of libraries. Could these data types go into a separate library, at least to start with?

**Comment by xgrommx** [comment]
@hdgarrood do u mean smth like `purescript-profunctor-extra`?
Also I think all things in `purescript-profunctor-lens` regarding `Profunctor` should be moved to `purescript-profunctor`
I mean `Wander`, `Star`, `Costar` etc

**Comment by hdgarrood** [comment]
Yeah, I think creating a separate library called `purescript-profunctor-extra` might not be a bad option. I don't have a huge amount of experience with profunctors and lenses though, so maybe one of the other maintainers might like to weigh in? I don't have an opinion on `Wander`, `Star`, `Costar` etc either.

**Pull request closed without merging**